### PR TITLE
Format code

### LINKED ISSUE #1589: [8.0.x] Code formatting
We have different styles floating around the code right now.
Let's use Scalafmt to make things consistent.

@fommil says a new sbt plugin is going to be available in the near future, so we can probably wait for that.

This pretty much sums up step 2:

> then we can fight over settings, I have some pretty aggressive settings

This is @fommil's set of preferences:

```
align = most
continuationIndent.defnSite = 2
assumeStandardLibraryStripMargin = true
docstrings = JavaDoc
lineEndings = preserve
includeCurlyBraceInSelectChains = false
danglingParentheses = true
spaces {
  inImportCurlyBraces = true
  inByNameTypes = false
}
optIn.annotationNewlines = true

rewrite.rules = [SortImports, RedundantBraces]
```

This is what I currently use:
```
maxColumn = 120
align.openParenCallSite = false
align.openParenDefnSite = false
align.ifWhileOpenParen = false
continuationIndent.defnSite = 2
continuationIndent.extendSite = 2
optIn.selfAnnotationNewline = false
optIn.configStyleArguments = false
spaces.afterTripleEquals = true
spaces.inImportCurlyBraces = true
danglingParentheses = true
includeCurlyBraceInSelectChains = false
importSelectors = singleLine
```

Related to #1464

### PR DESCRIPTION
**Pull request by NeQuissimus**
Closes #1589 

The build will fail if the code is not formatted correctly.
I mostly used a set of preferences from @fommil .

### CODE REVIEW

**Comment by fommil** [comment]
Nice! The PR is too big to see, are we enforcing formatting in CI now?

**Comment by NeQuissimus** [comment]
@fommil There is no formatting right now, so no. Post-PR we will.

Should we make "this will be much better very soon" a step 2 PR or wait for it here?

**Comment by fommil** [comment]
Step 2

**Comment by NeQuissimus** [comment]
Ping... I re-did this since the conflicts had accumulated :)
I also split into multiple commits to make a bit more sense...

File: .travis.yml [hunk 1]
```diff
@@ -12,4 +12,4 @@ install:
   - gem install sass
   - gem install jekyll -v 3.6.2
 script:
-  - sbt ++$TRAVIS_SCALA_VERSION test example/makeMicrosite
+  - sbt ++$TRAVIS_SCALA_VERSION scalafmtCheck test example/makeMicrosite
```
Comment (fommil) [inline .travis.yml]: btw, I don't think this does test configs or the sbt / project files.

Comment (xuwei-k) [reply .travis.yml]: https://github.com/scalaz/scalaz/pull/1649

**Pull request merged**